### DISCUSSION TITLE
How to build a cross-compiled Docker image?

### DISCUSSION THREAD

**Discussion by nitnelave** [cross-rs/cross#612] [Q&A]
Hi,

I'm trying to get a cross-compiled Docker image for Raspberry Pi alpine (armv7-unknown-linux-musleabihf).

I can build my project from outside docker with `cross build --target=armv7-unknown-linux-musleabihf`, but how can I get it working inside a Dockerfile?

Things I tried:
  - Running without cross
    - I'm running into issues due to the mismatch between the distribution's musl and Rust's musl.
  - Running cross inside docker
    - I would need to mount the docker socket, but it's not possible to mount things during the build.
  - Starting from the rustembedded/cross docker image, installing rust, and building with `cargo build`
    - I'm running into the same issues as the first case.

The musl issues manifest themselves as:
```
arm-linux-musleabi-ld: /app/target/armv7-unknown-linux-musleabi/release/deps/liblibsqlite3_sys-02b4bc2b68dd5f0a.rlib(sqlite3.o): in function `unixRandomness':
sqlite3.c:(.text.unixRandomness+0xec): undefined reference to `__time64'
```

How can I build a cross-compiled image? Which way seems most promising?

**Comment by Emilgardis** [comment]
for cross inside docker, see https://github.com/rust-embedded/cross#docker-in-docker

does that help?

**Reply by nitnelave** [reply]
I wish, but you can't mount the docker socket when _building_ an image, only when _running_ it.
